selectors/views/threads.ts

```diff
@@ -72,3 +91,87 @@ export const isThreadOpen = (state: GlobalState, threadId: $ID<UserThread>): boo
 export const isThreadManuallyUnread = (state: GlobalState, threadId: $ID<UserThread>): boolean => {
     return state.views.threads.manuallyUnread[threadId] || false;
 };
+
+// Returns a selector that, given the state and an object containing an array of postIds and an optional
+// timestamp of when the channel was last read, returns a memoized array of postIds interspersed with
+// day indicators and an optional new message indicator.
+export function makeFilterRepliesAndAddSeparators() {
+    const getPostsForIds = makeGetPostsForIds();
+
+    return createIdsSelector(
+        'makeFilterPostsAndAddSeparators',
+        (state: GlobalState, {postIds}: PostFilterOptions) => getPostsForIds(state, postIds),
+        (_state: GlobalState, {lastViewedAt}: PostFilterOptions) => lastViewedAt,
+        (_state: GlobalState, {showDate}: PostFilterOptions) => showDate,
+        (_state: GlobalState, {collapsedThreads}: PostFilterOptions) => collapsedThreads,
```

Comment:
Perhaps instead of having a separate option for `collapsedThreads`, the caller could just omit `lastViewedAt`?